src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2317,6 +2322,16 @@ int journal_file_append_entry(
                 boot_id = &_boot_id;
         }
 
+        r = sd_id128_get_machine(&_machine_id);
+        if (r < 0) {
+                /* If the machine ID is not initialized yet, handle gracefully */
+                if (!IN_SET(r, -ENOENT, -ENOMEDIUM, -ENOPKG))
```

Comment:
Do we need a `ERRNO_IS_XXX()` helper for this?

Reply:
→ https://github.com/systemd/systemd/pull/26284